# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/hotfix-mgag200-bigendian.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- linux-6.17/drivers/gpu/drm/mgag200/mgag200_mode.c	2025-09-28 23:39:22.000000000 +0200
+++ linux-6.17/drivers/gpu/drm/mgag200/mgag200_mode.c	2025-11-30 14:25:20.932807256 +0100
@@ -655,6 +655,20 @@
 	else
 		mgag200_crtc_fill_gamma(mdev, format);
 
+#ifdef __BIG_ENDIAN
+	/* Big-endian byte-swapping */
+	switch (format->format) {
+	case DRM_FORMAT_RGB565:
+		WREG32(MGAREG_OPMODE, 1 << 16);
+		break;
+	case DRM_FORMAT_XRGB8888:
+		WREG32(MGAREG_OPMODE, 2 << 16);
+		break;
+	default:
+		break;
+	}
+#endif
+
 	mgag200_enable_display(mdev);
 }
 
